### PR TITLE
Components should always contain at least one entry

### PR DESCRIPTION
**Pull request by Luthaf**
----
:books: Documentation preview :books:: https://equistore--232.org.readthedocs.build/en/232/

### CODE REVIEW

File: python/src/equistore/operations/equal_metadata.py [hunk 1]
```diff
@@ -240,6 +240,9 @@ def _check_blocks(a: TensorBlock, b: TensorBlock, props: List[str], fname: str):
                 if not (c1.names == c2.names):
                     raise ValueError(err_msg + err_msg_names)
 
+                if not (len(c1) == len(c2)):
+                    raise ValueError(err_msg + err_msg_1)
+
```
Comment (jwa7) [inline python/src/equistore/operations/equal_metadata.py:245]: I'm not sure I see the point of this check. Doesn't the check on line 246 do the same thing?
```py
if not np.all(c1 == c2):
    raise ValueError(err_msg + err_msg_1)
```

**Review by jwa7 — COMMENTED** [review]
Please could you briefly explain what the aim of these changes are - not sure I quite see it? (Perhaps better in person)

File: python/src/equistore/operations/equal_metadata.py (hunk 1 above)
Comment (Luthaf) [reply python/src/equistore/operations/equal_metadata.py:245]: The line below does check the same thing, but creates a warning about elementwise comparison in numpy.

**Comment by Luthaf** [comment]
This is mainly about consistency and restricting the number of possible states a TensorBlock can be in. I see no use-case for having 0 components (if you think of one, I'm happy to revisit!). This is different from samples and properties where having 0 of them can be useful/happen through slicing/...

**Comment by jwa7** [comment]
Discussion in person: understood and approved

**Pull request merged**